### PR TITLE
allow `docker inspect -f {{Created}}` to fail

### PR DESCRIPTION
**Pull request by mattnworb**
will fix https://github.com/spotify/docker-gc/issues/209

### CODE REVIEW

File: docker-gc [hunk 1]
```diff
@@ -275,10 +275,17 @@ $DOCKER images --no-trunc --format "{{.ID}}" --filter dangling=true >> images.al
...
-    ELAPSED=$(elapsed_time $CREATED)
-    if [[ $ELAPSED -gt $GRACE_PERIOD_SECONDS ]]; then
-        echo $line >> images.reap.tmp
+    retval=$?
+    set -o errexit
+    if [ $retval -eq 0 ]; then
+        ELAPSED=$(elapsed_time $CREATED)
+        if [[ $ELAPSED -gt $GRACE_PERIOD_SECONDS ]]; then
+            echo $line >> images.reap.tmp
+        fi
```
Comment (lavanyachennupati) [inline docker-gc]: iiuc, in case the `docker inspect` fails, though the script doesn't abort, it wouldn't GC the docker images right?

Comment (mattnworb) [reply docker-gc]: The image that `docker inspect ...` failed for won't be considered for removal, correct. But my hope is this change allows the script to continue to remove other images that don't return this error.

File: docker-gc [hunk 2]
```diff
@@ -275,10 +275,17 @@ $DOCKER images --no-trunc --format "{{.ID}}" --filter dangling=true >> images.al
...
     fi
 done
 comm -23 images.reap.tmp images.used | grep -E -v -f $EXCLUDE_IDS_FILE > images.reap || true
```
Comment (lavanyachennupati) [inline docker-gc]: Specifically would we reap out images that are not at least "GRACE_PERIOD_SECONDS" ago if the `docker inspect` fails?
Is the only way to fix this currently to wipe out `/var/lib/docker `?

Comment (mattnworb) [reply docker-gc]: Yes those images would not be garbage collected, since the script can't finish interrogating` docker` to determine their age.

**Pull request merged**